lib/david/resource_discovery.rb

```diff
@@ -70,15 +103,19 @@ def filter(link)
     end
 
     def include_route?(route)
-      !(route[0] =~ /\A\/(assets|rails)/)
+      !(route[0] =~ /\A\/(assets|rails|cable)/)
     end
 
     def routes
-      Rails.application.routes.routes.map do |route|
+      Rails.application.routes.routes.select { |route|
+        route.defaults[:coap]
```

Comment:
Actually, I am not convinced, it should be necessary to add routes explicitly although it could be helpful in a bigger app that focuses primarily in http.

I think, I would like it to be configurable with all routes bring included by default.

Reply:
Yes, so I didn't want my ActiveScaffold routes showing up in the CoAP interface...   I agree that a configuration that includes all routes would be a good thing.   Where should I put that configruation?
Also I was initially trying to annotate the routes using the discoverable clause in the controller, and I think that should be possible as well.  I just didn't find an elegant way yet.

Reply:
Rails only configuration options are perfectly fine in `lib/david/railties/config.rb`.

Reply:
A configuration option in lib/david/railties/config.rb would apply to all applications... I'm thinking that it belongs in the application's config/routes.rb.  (Sorry for coming back to this after such a long time. I've been in OpenSSL DTLS hell. It works now though)